packages/js/components/src/form/form.tsx

```diff
@@ -123,6 +122,43 @@ export type ConsumerInputProps< Values > = {
 	sanitize?: ( value: Values[ keyof Values ] ) => Values[ keyof Values ];
 };
 
+type StateAndHelpers< Values > = {
+	values: Values;
+	errors: FormErrors< Values >;
+	touched: { [ P in keyof Values ]?: boolean };
+	isDirty: boolean;
+	isValidForm: boolean;
+	setTouched: React.Dispatch<
+		React.SetStateAction< { [ P in keyof Values ]?: boolean | undefined } >
+	>;
+	setValue: ( name: string, value: Values[ keyof Values ] ) => void;
+	setValues: ( valuesToSet: Values ) => void;
+	handleSubmit: () => unknown;
+	getCheckboxControlProps: (
+		name: string,
+		inputProps?: ConsumerInputProps< Values >
+	) => CheckboxProps< Values, Values[ keyof Values ] >;
+	getInputProps: < P extends keyof Values >(
+		name: P,
+		inputProps?: ConsumerInputProps< Values >
+	) => InputProps< Values, Values[ P ] >;
```

Comment:
Using the `P` here to make sure that `getInputProps` returns the correct value for the passed in parameter.
This means that `getInputProps('name')` will correctly return a string and `getInputProps('count')` will return a `number` ( dependent on the `Values` type of-course ).